### ISSUE TITLE
SPDX-Online-Tools shows error message for License in Japanese

### ISSUE THREAD

**Issue by mandel59** [spdx/spdx-online-tools#267]
I've submitted the license text from https://creativecommons.org/licenses/by-sa/2.1/jp/legalcode, and I've got the following message:

>Unexpected error, please email the SPDX technical workgroup that the following error has occurred: Traceback (most recent call last): File "/spdx/src/app/views.py", line 127, in submitNewLicense matchingIds, matchingType = utils.check_spdx_license(licenseText) File "/spdx/src/app/utils.py", line 509, in check_spdx_license licenseText = unicode(licenseText.decode('string_escape'), 'utf-8') UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-16: ordinal not in range(128) 

![image](https://user-images.githubusercontent.com/1079715/100558353-6a7cc100-32f1-11eb-9919-87f500c51e25.png)

**Comment by goneall** [comment]
@rtgdk @Ugtan Any ideas on how to resolve this issue?

**Comment by rtgdk** [comment]
This happens because we are trying to pass a Unicode string containing non-English characters (characters beyond 128) to something that expects an ASCII bytestring and then trying to encode it again to unicode with utf-8.
I am not sure why this didn't work already since we have set default encoding for the files as utf-8 (first line of every python file). There are a couple of ways to fix this:
1. encode the license text to utf-8 before passing it to unicode(...).
2. Encode by ignoring errors 
`unicode(..., errors='ignore')`
3. I just found out these django utility function in the module 
`from django.utils.encoding import smart_str, smart_unicode` which handles this encoding/decoding.
4. Setting the system encoding to utf-8 by default -

> import sys
> sys.setdefaultencoding('utf-8')

@Ugtan Let me know what you think.

**Comment by rtgdk** [comment]
Also, this error is specific to python2 since unicode/str differentiation is removed in python3.

**Comment by Ugtan** [comment]
@rtgdk You are absolutely right. This error is due to the reason mentioned by you above. And, yeah it is limited to python 2 only. Thumbs up for the point 3, wasn't aware of it.

**Comment by goneall** [comment]
Setting a breakpoint and looking at the "string" in the debugger shows that the string is already a unicode class encoded in UTF-8.

The existing code looks like it is trying to do 2 things:
1. make sure it is a unicode type by encoding at UTF-8
2. unescape any escaped values using the `string_escape` [Python codec](https://docs.python.org/2/library/codecs.html#standard-encodings)

I added a pull request to only convert if it not already a unicode class.  This takes care of the first item above, but it doesn't take care of the second item.

It doesn't look like the strings coming into this function area actually escaped.  I ran a test to see if the unescaping changed a simple string with quotes and a linefeed and it produced this result:

```
(Pdb) licenseText
u'This is "some text\'s" to test.\r\nThis is the second line.'
(Pdb) unicode(licenseText.decode('string_escape'), 'utf-8')
u'This is "some text\'s" to test.\r\nThis is the second line.'
(Pdb)
```
@Ugtan @rtgdk Can you verify that we do not need to unescape the string?  If we don't need to escape, PR #268 should resolve this issue.

**Comment by rtgdk** [comment]
I think we would be fine without string_escape since the values the django form would be receiving would be in the form above. Also decoding with string_escape returns a string (ascii) which will not work for non-English characters. There should be a somewhat neater way to do this which handles both the cases, but for now PR looks fine to fix this.

**Comment by rtgdk** [comment]
Btw for unicodes we can use unicode_escape to handle escaping, but I think let's not do that right now.

**Comment by goneall** [comment]
@rtgdk Thanks for the review.

I just grepped the source and found two more `str.decode('string_escape')` calls in views.py:

- https://github.com/spdx/spdx-online-tools/blob/5e6a14c38c416a2a48c1a752f03afe8f57352070/src/app/views.py#L128
- https://github.com/spdx/spdx-online-tools/blob/5e6a14c38c416a2a48c1a752f03afe8f57352070/src/app/views.py#L159

I'll investigate whether these could cause the same issue and if they should just be removed before merging in the PR.

**Comment by goneall** [comment]
The other 2 calls to decode did indeed cause an issue.  I verified that the strings being passed are already unicode and do not have escaped strings when entered from the UI.

I've removed them in the PR.

**Comment by goneall** [comment]
Resolved with PR #268 

This has now been deployed to production.

@mandel59 Thank you for the reporting the issue.  Please let let us know if you run into any further issues.
